client/app/bundles/course/learning-map/actions.js

```diff
@@ -0,0 +1,110 @@
+import CourseAPI from 'api/course';
+import { actionTypes } from './constants';
+
+export function fetchNodes() {
+  return (dispatch) => {
+    dispatch({ type: actionTypes.LOADING });
+
+    return CourseAPI.learningMap.index()
+      .then((response) => {
+        dispatch({
+          type: actionTypes.FETCH_LEARNING_MAP_SUCCESS,
+          nodes: response.data.nodes,
+          canModify: response.data.can_modify,
+        });
+      }).catch(() => {
+        dispatch({ type: actionTypes.FETCH_LEARNING_MAP_FAILURE });
```

Comment:
You can add failure notification here (Can refer to other part of the codebase)

`setNotification(failureMessage)(dispatch)`

Reply:
@zhuhanming Do you think we should standardise how we show all the error messages in this PR using the NotificationBar component?

Reply:
Yep would be good.

Reply:
But maybe don't need to be too strict, since I think it will be standardised when we refactor the client side code out.

Reply:
Regarding the use of `NotificationBar`, I currently show success and failure messages using my `Dashboard` component. Would it be better if I shifted both to use the `NotificationBar` then? The learning map interactions (e.g. deleting a condition) will still be executed via interacting with the `Dashboard`

Reply:
Hmm then let's utilise your `Dashboard` instead.

We can relook into this later when migrating.